From 3536c8cf6de8ce094b7df2081f2223ac26213167 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Mon, 4 Feb 2019 18:39:32 +0200
Subject: [PATCH 0999/1051] net: mvpp2: fix missed spin_lock_init before
 spinlock use

Issue were found by "Lock debugging: prove locking correctness"
kernel tool.
spin_lock_init should be called before first spinlock use.
Issue didn't caused any real issue, but considered as incorrect
spinlock use.

Change-Id: I2dc846351001162152131a47042e26ad5da2c5ac
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/3444
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Reviewed-by: Marcin Wojtas <Marcin.Wojtas@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index fa4ed434acca..8667ee3f9599 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -6348,6 +6348,12 @@ static int mvpp2_port_probe(struct platform_device *pdev,
 		}
 	}
 
+	/* Init TX locks and bm locks */
+	for (i = 0; i < MVPP2_MAX_THREADS; i++) {
+		spin_lock_init(&port->bm_lock[i]);
+		spin_lock_init(&port->tx_lock[i]);
+	}
+
 	return 0;
 
 err_phylink:
-- 
2.17.1

